src/services/usersService.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+class UserModel {
+  constructor() {
+    this.users = [];
+  }
+
+  reset() {
+    this.users = [];
+  }
+
+  getAll() {
+    return this.users;
+  };
+
+  getById(userId) {
+    return this.users.find(user => user.id === userId) || null;
+  }
+
+  addUser(name) {
+    const maxId = this.users.length
```

Comment:
Use `const maxId = this.users.reduce((max, user) => Math.max(max, user.id), -1);` instead of using `Math.max` with spread operator over `map` to simplify and optimize the code.